### PR TITLE
[19.09] Fix data_manager_manual with new data tables + cleanup and fixes

### PR DESCRIPTION
**Pull request by mvdbeek**
Builds on #8792 by @abretaud 

- Fixes and unifies DM loading
- More useful traceback when loading DM failed
- Implement get_tool_relative_path, needed for running data_manager_manual without install db
- Test data_manager_manual in an integration test

### CODE REVIEW

**Comment by mvdbeek** [comment]
I think this is ready for review now.

File: lib/galaxy/model/tool_shed_install/__init__.py [hunk 1]
```diff
@@ -79,12 +79,17 @@ def can_reinstall_or_activate(self):
...
+    @property
+    def shed_config_filename(self):
         shed_config_filename = None
         if self.metadata:
             shed_config_filename = self.metadata.get('shed_config_filename', shed_config_filename)
         return shed_config_filename
 
+    @shed_config_filename.setter
+    def shed_config_filename(self, value):
+        self.metadata['shed_config_filename'] = value
```
Comment (nsoranzo) [inline lib/galaxy/model/tool_shed_install/__init__.py:88]: Should we make sure that `self.metadata` is not `None` by using `self.metadata = metadata or {}` in the class `__init__()` method (line 45)?

Then so many methods can be simplified.

File: lib/galaxy/tools/data_manager/manager.py [hunk 2]
```diff
@@ -75,8 +71,8 @@ def load_manager_from_elem(self, data_manager_elem, tool_path=None, add_manager=
             if e.errno == errno.ENOENT:
                 # File does not exist
                 return None
-        except Exception as e:
-            log.error("Error loading data_manager '%s':\n%s" % (e, util.xml_to_string(data_manager_elem)))
+        except Exception:
+            log.error("Error loading data_manager", exc_info=True)
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_manager/manager.py]:
```suggestion
            log.exception("Error loading data_manager")
```

File: lib/galaxy/tools/data_manager/manager.py [hunk 3]
```diff
@@ -143,65 +139,31 @@ def load_from_element(self, elem, tool_path):
...
         self.version = elem.get('version', self.version)
-        tool_shed_repository_id = None
+        tool_shed_repository = None
         tool_guid = None
 
         if path is None:
             tool_elem = elem.find('tool')
             assert tool_elem is not None, "Error loading tool for data manager. Make sure that a tool_file attribute or a tool tag set has been defined:\n%s" % (util.xml_to_string(elem))
-            path = tool_elem.get("file", None)
+            path = tool_elem.get("file", path)
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_manager/manager.py]:
```suggestion
            path = tool_elem.get("file")
```

File: lib/galaxy/model/tool_shed_install/__init__.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/model/tool_shed_install/__init__.py:88]: I think that should work, I'll give it a try

File: lib/galaxy/tools/data_manager/manager.py [hunk 4]
```diff
@@ -143,65 +139,31 @@ def load_from_element(self, elem, tool_path):
...
-                    shed_conf = self.data_managers.app.toolbox.get_shed_config_dict_by_filename(shed_conf_file, None)
-                    if shed_conf:
-                        tool_path = shed_conf.get("tool_path", tool_path)
+            tool_shed_repository = self.data_managers.app.toolbox.get_tool_repository_from_xml_item(tool_elem, path)
+            self.tool_shed_repository_info_dict = dict(tool_shed=tool_shed_repository.tool_shed,
+                                                       name=tool_shed_repository.name,
+                                                       owner=tool_shed_repository.owner,
+                                                       installed_changeset_revision=tool_shed_repository.installed_changeset_revision)
+            # use shed_conf_file to determine tool_path
+            shed_conf_file = elem.get("shed_conf_file", None)
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_manager/manager.py]:
```suggestion
            shed_conf_file = elem.get("shed_conf_file")
```

**Comment by bgruening** [comment]
@mvdbeek here are now some conflicts.

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/tools/__init__.py [hunk 5]
```diff
@@ -520,7 +520,6 @@ def tool_shed_repository(self):
                                                             name=self.repository_name,
                                                             owner=self.repository_owner,
                                                             installed_changeset_revision=self.installed_changeset_revision,
-                                                            repository_id=self.repository_id,
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py:523]: Not sure I understand this change.

Comment (mvdbeek) [reply lib/galaxy/tools/__init__.py:523]: If you specify a repository_id we'll not be able to use ToolConfRepository items, which don't have an id. We should only provide the repository_id when working in the galaxy_install module.

**Pull request merged**